src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowEdit.cs

```diff
@@ -38,6 +38,11 @@ public abstract class RowEditBase : IComparable<RowEditBase>
         /// <param name="associatedMetadata">Metadata provider for the object to edit</param>
         protected RowEditBase(long rowId, ResultSet associatedResultSet, EditTableMetadata associatedMetadata)
         {
+            if (!associatedMetadata.HasExtendedProperties)
+            {
+                throw new ArgumentException(SR.EditDataMetadataNotExtended);
```

Comment:
Yes, this will be caught by the Handle*Requests (in our case HandleUpdateCellRequest) and returned as an error.